src/client/content_cao.cpp

```diff
@@ -563,6 +566,21 @@ void GenericCAO::removeFromScene(bool permanent)
 	}
 }
 
+void GenericCAO::setSceneNodeMaterial()
+{
+	scene::ISceneNode *node = getSceneNode();
+
+	node->setMaterialFlag(video::EMF_LIGHTING, false);
+	node->setMaterialFlag(video::EMF_BILINEAR_FILTER, false);
+	node->setMaterialFlag(video::EMF_FOG_ENABLE, true);
+	node->setMaterialType(m_material_type);
+
+	if (m_enable_shaders) {
+		node->setMaterialFlag(video::EMF_GOURAUD_SHADING, false);
+		node->setMaterialFlag(video::EMF_NORMALIZE_NORMALS, true);
+	}
+}
```

Comment:
For code quality, it'd be nice to turn this into a lambda inside `addToScene`.
Like this:
```cpp
auto setSceneNodeMaterial = [=] (scene::ISceneNode *node) {
	node->setMaterialFlag(video::EMF_LIGHTING, false);
	node->setMaterialFlag(video::EMF_BILINEAR_FILTER, false);
	node->setMaterialFlag(video::EMF_FOG_ENABLE, true);
	node->setMaterialType(m_material_type);

	if (m_enable_shaders) {
		node->setMaterialFlag(video::EMF_GOURAUD_SHADING, false);
		node->setMaterialFlag(video::EMF_NORMALIZE_NORMALS, true);
	}
};
```

Reply:
Done